hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java

```diff
@@ -910,6 +910,27 @@ public void testCreateBucketWithECReplicationConfig() throws Exception {
     }
   }
 
+  @Test
+  public void testPutKeyOnBucketWithECReplicationConfig() throws Exception {
+    final String volumeName = "volume1000";
+    getVolume(volumeName);
+    String[] args =
+        new String[] {"bucket", "create", "/volume1000/bucket0", "-t", "EC",
+            "-r", "rs-3-2-1024k"};
+    execute(ozoneShell, args);
+
+    args = new String[] {"key", "put", "/volume1000/bucket0/newECKey1",
+        testFilePathString};
+    execute(ozoneShell, args);
+    OzoneVolume volume =
+        cluster.getClient().getObjectStore().getVolume(volumeName);
+    OzoneBucket bucket = volume.getBucket("bucket0");
+    try (OzoneOutputStream out = bucket.createKey("newECKey1", 1024)) {
+      Assert.assertTrue(out.getOutputStream().getClass().getName()
```

Comment:
nit: we can use asserEquals here on the class instances

Reply:
changed this assert to check types now on KeyDetails.